apps/explorer/src/pages/transaction-result/transaction-summary/ObjectChanges.tsx

```diff
@@ -0,0 +1,335 @@
+                                    <Text
+                                        variant="pBody/medium"
+                                        color="steel-dark"
+                                    >
+                                        {type === 'Shared' ? 'Shared' : 'Owner'}
+                                    </Text>
+                                    {type === 'AddressOwner' && (
+                                        <AddressLink address={locationId} />
+                                    )}
+                                    {type === 'ObjectOwner' && (
+                                        <ObjectLink objectId={locationId} />
+                                    )}
+                                    {type === 'Shared' && (
+                                        <Text
+                                            variant="pBody/medium"
+                                            color="hero-dark"
+                                        >
+                                            {locationId}
+                                        </Text>
+                                    )}
+                                </div>
+                            )
+                        }
+                    >
+                        <TransactionCardSection
+                            expanded={expanded}
+                            setExpanded={setExpanded}
+                            title={
+                                <Text
+                                    variant="body/semibold"
+                                    color="success-dark"
+                                >
+                                    {title}
+                                </Text>
+                            }
+                        >
+                            {changes.map(({ objectId, objectType }) => (
+                                <ObjectDetail
+                                    key={objectId}
+                                    objectId={objectId}
+                                    objectType={objectType}
+                                />
+                            ))}
+                        </TransactionCardSection>
+                    </TransactionCard>
+                );
+            })}
+        </>
+    );
+}
+
+function groupByOwner(objectSummaryChanges: any) {
+    if (!objectSummaryChanges) {
+        return {};
+    }
+
+    return objectSummaryChanges?.reduce(
+        (mapByOwner: Record<string, any>, change: any) => {
+            const owner = change?.owner;
+
+            let key = '';
+            if ('AddressOwner' in owner) {
+                key = `AddressOwner${OWNER_GROUP_DIVIDER}${owner.AddressOwner}`;
+            } else if ('ObjectOwner' in owner) {
+                key = `ObjectOwner${OWNER_GROUP_DIVIDER}${owner.ObjectOwner}`;
+            } else if ('Shared' in owner) {
+                key = `Shared${OWNER_GROUP_DIVIDER}${owner.Shared?.initial_shared_version}`;
```

Comment:
shared version isn't really the right thing to use, you need to use the shared object ID.